src/transformers/generation/utils.py

```
@@ -4439,42 +4430,43 @@ def assisted_decoding(
-                    # `new_token_len` can be 1 or 2 (next token in assistant + last token picked by the larger model)
-                    new_token_len = candidate_input_ids.shape[1] - prev_seq_len
-                    assist_inputs = candidate_input_ids[:, -new_token_len:]
-                    # TODO (joao): make it compatible with models that use unconventional fwd pass logic, like blip2
-                    if assistant_model.config.is_encoder_decoder:
-                        assistant_model_outputs = assistant_model(
-                            decoder_input_ids=assist_inputs,
-                            past_key_values=model_kwargs["assistant_past_key_values"],
-                            encoder_outputs=model_kwargs["assistant_encoder_outputs"],
-                        )
-                    else:
-                        assistant_model_outputs = assistant_model(
-                            assist_inputs,
-                            past_key_values=model_kwargs["assistant_past_key_values"],
-                        )
-                else:
-                    if assistant_model.config.is_encoder_decoder:
-                        assistant_model_outputs = assistant_model(
-                            decoder_input_ids=candidate_input_ids,
-                            encoder_outputs=model_kwargs["assistant_encoder_outputs"],
-                        )
-                    else:
-                        assistant_model_outputs = assistant_model(candidate_input_ids)
+                assistant_inputs = assistant_model.prepare_inputs_for_generation(
+                    candidate_input_ids,
+                    attention_mask=assistant_attention_mask,
+                    decoder_attention_mask=assistant_decoder_attention_mask,
+                    encoder_outputs=assistant_encoder_outputs,
+                    past_key_values=model_kwargs.get("assistant_past_key_values", None),
+                )
```

request: this change should work due to a recent PR in the models' prepare_inputs_for_generation, and the input_ids/decoder_input_ids in assistant_inputs should have .shape[-1] equal to 1 or 2. Can we raise an exception here if it is not the case?

(there may be silent failures in custom code on the hub if we don't add the exception)